src/components/routers/GroupRouter.tsx

```diff
@@ -0,0 +1,124 @@
+import * as React from 'react';
+import { v4 as generateUUID } from 'uuid';
+import { Node, SwitchRouter, WaitType } from '../../flowTypes';
+import { SearchResult } from '../../services/ComponentMap';
+import { endpointsPT } from '../../providers/ConfigProvider/propTypes';
+import { FormProps } from '../NodeEditor/NodeEditor';
+import GroupElement, { GroupElementProps } from '../form/GroupElement';
+import { hasSwitchRouter, hasWait, resolveExits } from './SwitchRouter';
+import TextInputElement from '../form/TextInputElement';
+import { GROUP_LABEL, GROUPS_OPERAND } from './constants';
+
+import { instructions } from './SwitchRouter.scss';
+import { CaseElementProps } from '../form/CaseElement';
+
+type GroupRouterProps = Partial<FormProps>;
+
+export const extractGroups = ({ exits, router }: Node): SearchResult[] =>
+    (router as SwitchRouter).cases.map(kase => {
+        const resultName = exits.reduce((result, { name, uuid }) => {
+            if (uuid === kase.exit_uuid) {
+                result += name;
+            }
+            return result;
+        }, '');
+        return { name: resultName, id: kase.arguments[0] };
+    });
+
+export const groupSplitExistsAtNode = (node: Node) =>
+    hasSwitchRouter(node) && hasWait(node, WaitType.group);
+
+export const toCases = (groups: SearchResult[] = []): CaseElementProps[] =>
```

Comment:
this helper maps groups (as `SearchResult` objects) to `CaseElementProps` objects so we can pass them to `SwitchRouter`'s `resolveExits` method. Not sure if we want to reuse that method but it's working here so far 🤷‍♂️

Reply:
Feels a bit odd since if I recall resolveExits' primary job is to merge like-named exit cases. I suppose it also tacks on the catch all case, maybe we break that bit out so we aren't trying to munge exits when it's not necessary? Just a thought going forward, this is fine for this PR.